clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2114,6 +2154,17 @@ class SyclKernelBodyCreator : public SyclKernelFieldHandler {
     SemaRef.PushFunctionScope();
     KernelBodyTransform KBT(MappingPair, SemaRef);
     Stmt *NewBody = KBT.TransformStmt(FunctionBody).get();
+
+    if (hasSyclKernelHandlerArg(KernelCallerFunc)) {
+      // Factor this out. Repetitive code.
+      ParmVarDecl *KernelHandlerParam = KernelCallerFunc->getParamDecl(1);
```

Comment:
So I'd rather this do a bit of a 'find' rather than assume its the 2nd parameter.  Same with the 'has' function, but find it in the list by looping instead of assuming.

Reply:
Alright. Will do!

Reply:
Done!